### PR TITLE
Issue #82 solved: Renaming project option added in project explorer.

### PR DESCRIPTION
**Pull request by anjalijaiswal08**

### CODE REVIEW

File: src/subcircuit/convertSub.py [hunk 1]
```diff
@@ -38,12 +38,12 @@ def createSub(self):
...
+                    'The subcircuit does not contain any Kicad netlist file'
+                    + 'for conversion.')
                 self.msg.setWindowTitle("Error Message")
         else:
             self.msg = QtGui.QErrorMessage(None)
             self.msg.showMessage(
-                'Please select the subcircuit first. You can either create \
-                new subcircuit or open existing subcircuit')
+                'Please select the subcircuit first. You can either create'
+                + 'new subcircuit or open existing subcircuit')
```
Comment (sunilshetye) [inline src/subcircuit/convertSub.py]:
```suggestion
                + ' new subcircuit or open existing subcircuit')
```
require space between create and new

File: src/frontEnd/ProjectExplorer.py [hunk 2]
```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
...
-                updatedProjectFiles
-            )
+            self.workspace = self.obj_appconfig.default_workspace['workspace']
+            self.newBaseFileName = str(self.newBaseFileName).rstrip().lstrip()
+            self.projDir = os.path.join(self.workspace, str(self.newBaseFileName))
 
-            # save project_explorer dictionary on disk
-            json.dump(self.obj_appconfig.project_explorer,
-                      open(self.obj_appconfig.dictPath, 'w'))
+            if self.newBaseFileName == "":
```
Comment (sunilshetye) [inline src/frontEnd/ProjectExplorer.py]:
```suggestion
            if newBaseFileName == "":
```
don't use self. for local variables

File: src/frontEnd/ProjectExplorer.py [hunk 3]
```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
...
+            - If project names is not empty.
+            - Project name does not contain spaces between them.
+            - Project name is different between what it was earlier.
+            - Project name should not exist.
+
+        And after project name is changed it recreates the project explorer tree.
+        """
+        self.indexItem = self.treewidget.currentIndex()
+        self.baseFileName = str(self.indexItem.data())
+        self.newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
```
Comment (sunilshetye) [inline src/frontEnd/ProjectExplorer.py]:
```suggestion
        newBaseFileName, ok = QtGui.QInputDialog.getText(self, 'Rename Project', 'Project Name:',
```
don't use self. for local variables

File: src/frontEnd/ProjectExplorer.py [hunk 4]
```diff
@@ -226,51 +230,101 @@ def refreshProject(self):
...
+
+            elif self.baseFileName == self.newBaseFileName:
+                print("Project name has to be different")
+                print("==================")
+                msg = QtGui.QErrorMessage(self)
+                msg.showMessage('The project name has to be different')
+                msg.setWindowTitle("Error Message")
+
+            else:
+                self.reply = self.obj_validation.validateNewproj(str(self.projDir))
```
Comment (sunilshetye) [inline src/frontEnd/ProjectExplorer.py]:
```suggestion
                reply = self.obj_validation.validateNewproj(str(self.projDir))
```
don't use self. for local variables

**Comment by anjalijaiswal08** [comment]
This pull request had a lot of unwanted commits, so removed.

**Pull request closed without merging**